app/scripts/lib/config-loader.js

```diff
@@ -58,16 +56,16 @@ define([
         // did not receive the cookie.
       }
 
-      self._configPromise = self._xhr.getJSON('/config')
-        .fail(function (jqXHR) {
+      this._configPromise = this._xhr.getJSON('/config')
+        .fail((jqXHR) => {
           throw ConfigLoader.Errors.normalizeXHRError(jqXHR);
         });
 
-      return self._configPromise;
-    },
-  };
+      return this._configPromise;
+    }
+  }
 
-  var t = function (msg) {
+  let t = function (msg) {
```

Comment:
Speaking of `const`/`let` vs `var`, ESLint has a few other good rules for ES6 stuff:
- `no-var` - require `let` or `const` instead of `var`.
- `prefer-const` - suggest using `const` declaration for variables that are never modified after declared.

Of course, we'd need to make sure we don't break l10n, or whatever the recent issue was with `const` usage with the locale pipeline. But we could disable those ESLint rules for those specific files.